packages/maskbook/src/web3/hooks/useEtherTokenDetailed.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import type { AsyncStateRetry } from 'react-use/lib/useAsyncRetry'
+import { CONSTANTS } from '../constants'
+import { EtherTokenDetailed, EthereumTokenType } from '../types'
+import { useChainId } from './useChainState'
+import { useConstant } from './useConstant'
+
+export function useEtherTokenDetailed() {
+    const chainId = useChainId()
+    const ETH_ADDRESS = useConstant(CONSTANTS, 'ETH_ADDRESS')
+    return useMemo(() => {
+        return {
+            loading: false,
+            error: void 0,
+            value: {
+                type: EthereumTokenType.Ether,
+                address: ETH_ADDRESS,
+                chainId,
+                name: 'Ether',
+                symbol: 'ETH',
+                decimals: 18,
+            },
+        } as AsyncStateRetry<EtherTokenDetailed>
```

Comment:
This looks strange, why you're going to emulate the behavior of useAsync?

Reply:
For